### ISSUE TITLE
onListViewUpdated and onExecute don't function properly in SPFx extensions 

### ISSUE THREAD

**Issue by hamedy** [SharePoint/sp-dev-docs#4828]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

no matter how an item gets selected, the onListViewUpdated callback should be triggered with the current selected item passed through the event parameter. 

also the onExecute function should have the current selected item in the event parameter. 

#### Observed Behavior

-from the second time and on when you select an item, onListViewUpdated is not triggered unless if the selection is done through the radio button. If you just click on a row the callback is not triggered. 

-when you change your selection in anyway but through the radio button, when you click the extension button, the event parameter in onExecute contains the very fist item that was selected and not the current one. 

So for example opening the context menu (which would select the item too) and clicking on the extension button would not trigger the onListViewUpdated from second time and on and onExecute will have the wrong item in event parameter. 

I hope I'm doing something wrong, otherwise it's beyond me how this could not have been discovered in testing. We have deliveries for out customers soon and this would definitely put a stop to the whole project! We have flows and azure functions and an entire solution that all depends on this functionality to start the process. Should be able to get the selected item reliably. 

Looking forward to an answer and hopefully a solution. 
Hamed

**Comment by mrcpgn** [comment]
I have exactly the same issue.
onListViewUpdated method is not called if I have already selected an element and I select another one by clicking outside the checkbox, on the columns.

**Comment by CPritch** [comment]
Same issue on a client's tenant for me. Made for a less than fun bug hunt when issues were reported from testing.

Env:   SPO - General Release
SPFx: 1.9.1

**Comment by hamedy** [comment]
so we tried to work around this issue by rendering buttons on each row which would get a reference to current item and call a flow but then this happened: https://github.com/SharePoint/sp-dev-docs/issues/4656
onRenderCell not triggered on a random subset of items depending on how fast you scroll. 
We're basically stuck.

**Comment by jasonalex13** [comment]
Sound like the same issue I'm having on SPO (with SPFx generator 1.9.1)

- Works as expected first time (can select via any method: click anywhere in the row, right click etc)
- After that the radio button MUST be used to select a different row or the onExecute() event just returns the first selected item (OR the last item that was selected with the radio button)

**Comment by ltdu** [comment]
I think this is the same issue as described in #4769

**Comment by hamedy** [comment]
@andrewconnell can you try to escalate this issue please? We've suggested our customers to invest a lot to migrate from classic to modern because of better performance and SPFx customization possibilities and now we're looking like a fool since nothing works properly and project is on hold!

**Comment by CPritch** [comment]
Yeah, exact same boat as @hamedy for us. We are currently trying to release an ISO management library and quite rightly our client is terrified of the wrong documents being actioned.

**Comment by PeppeDotNet** [comment]
@VesaJuvonen can you please try to help us on this one? We have this issue in production environment, from more than 15 days. We have already opened premier support case (n. 17517628). Thank you in advance for your work.

**Comment by LukasNespor** [comment]
@PeppeDotNet the fix will be deployed soon. See https://github.com/SharePoint/sp-dev-docs/issues/4769#issuecomment-553021803

**Comment by andrewconnell** [comment]
DIdn't see this one... as this was submitted after #4769, this is technically a dupe so I'll close it. Please follow the other issue for status updates
